sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -117,7 +117,7 @@ private[sql] object DataFrame {
 @Experimental
 class DataFrame private[sql](
     @transient val sqlContext: SQLContext,
-    @DeveloperApi @transient val queryExecution: SQLContext#QueryExecution)
+    @DeveloperApi @transient var queryExecution: SQLContext#QueryExecution)
```

Comment:
We need to restrict this var to private, and provide a read only accessor to this var.